private/buf/bufsync/syncer.go

```diff
@@ -684,34 +684,35 @@ type moduleTarget struct {
 	expectedSyncPoint    string
 }
 
-// rebuildModule takes a built module, and rebuilds it with a new module identity.
-func rebuildModule(
+// renameModule takes a module, and rebuilds it with a new module identity.
+func renameModule(
 	ctx context.Context,
-	builtModule *bufmodulebuild.BuiltModule,
-	identityOverride bufmoduleref.ModuleIdentity,
+	baseModule *bufmodulebuild.BuiltModule,
+	newIdentity bufmoduleref.ModuleIdentity,
 ) (*bufmodulebuild.BuiltModule, error) {
-	if builtModule == nil {
-		return nil, errors.New("no built module to rebuild")
+	if baseModule == nil {
+		return nil, errors.New("no base module to rebuild")
 	}
-	if identityOverride == nil {
+	if newIdentity == nil {
 		return nil, errors.New("no new identity to apply")
 	}
-	if builtModule.ModuleIdentity().IdentityString() == identityOverride.IdentityString() {
-		// same identity, no need to rebuild anything
-		return builtModule, nil
+	if baseModule.ModuleIdentity() != nil &&
```

Comment:
I was missing this nil check.

Reply:
Already tested successfully locally. Syncing all branches:
- ~17min syncing the first time
- 16s the second time a couple of commits more